--- origsrc/evolution-ews-3.2.2/configure.ac	2011-11-14 03:04:55.000000000 -0600
+++ src/evolution-ews-3.2.2/configure.ac	2011-11-18 15:21:59.073195400 -0600
@@ -44,17 +44,29 @@ dnl ******************************
 dnl Check for Win32
 dnl ******************************
 
+AC_CANONICAL_HOST
 AC_MSG_CHECKING([for Win32])
 case "$host" in
+*-cygwin*)
+	os_win32=no
+	NO_UNDEFINED='-no-undefined'
+	SOCKET_LIBS=''
+	SOPFX='cyg'
+	SOEXT='.dll'
+	;;
 *-mingw*)
 	os_win32=yes
 	NO_UNDEFINED='-no-undefined'
 	SOCKET_LIBS='-lws2_32'
+	SOPFX='lib'
+	SOEXT='.dll'
 	;;
 *)
 	os_win32=no
 	NO_UNDEFINED='-no-undefined'
 	SOCKET_LIBS=''
+	SOPFX='lib'
+	SOEXT='.so'
 	;;
 esac
 
@@ -62,6 +74,8 @@ AC_MSG_RESULT([$os_win32])
 AM_CONDITIONAL(OS_WIN32, [test "x$os_win32" = "xyes"])
 AC_SUBST(NO_UNDEFINED)
 AC_SUBST(SOCKET_LIBS)
+AC_SUBST(SOPFX)
+AC_SUBST(SOEXT)
 
 dnl ***********************
 dnl Compiler Warning Flags
@@ -155,6 +169,11 @@ dnl ****************************
 PKG_CHECK_MODULES(EVOLUTION_PLUGIN, evolution-plugin-3.0 >= evo_minimum_version)
 
 dnl ****************************
+dnl Check for evolution mail
+dnl ****************************
+PKG_CHECK_MODULES(EVOLUTION_MAIL, evolution-mail-3.0 >= evo_minimum_version)
+
+dnl ****************************
 dnl Check for evolution shell 
 dnl ****************************
 PKG_CHECK_MODULES(EVOLUTION_SHELL, evolution-shell-3.0 >= evo_minimum_version)
--- origsrc/evolution-ews-3.6.1/src/camel/Makefile.am	2012-09-16 10:06:03.000000000 -0500
+++ src/evolution-ews-3.6.1/src/camel/Makefile.am	2012-12-20 03:38:41.902374600 -0600
@@ -39,7 +39,7 @@ noinst_HEADERS = \
 	camel-ews-transport.h \
 	$(NULL)
 
-libcamelews_la_LDFLAGS = -avoid-version $(NO_UNDEFINED)
+libcamelews_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 
 libcamelews_la_LIBADD = \
 	$(top_builddir)/src/server/libeews-1.2.la  \
--- origsrc/evolution-ews-3.6.1/src/configuration/Makefile.am	2012-09-16 10:06:03.000000000 -0500
+++ src/evolution-ews-3.6.1/src/configuration/Makefile.am	2012-12-20 03:35:13.624461800 -0600
@@ -46,8 +46,10 @@ module_ews_configuration_la_SOURCES = \
 module_ews_configuration_la_LIBADD = \
 	$(top_builddir)/src/server/libeews-1.2.la \
 	$(top_builddir)/src/utils/libewsutils.la \
+	$(top_builddir)/src/camel/libcamelews.la \
 	$(EVOLUTION_MAIL_LIBS) \
 	$(EVOLUTION_SHELL_LIBS) \
+	$(LIBBOOK_LIBS) \
 	$(NULL)
 
 module_ews_configuration_la_LDFLAGS = \
